rcl/test/rcl/test_guard_condition.cpp

```diff
@@ -127,6 +127,11 @@ TEST_F(
     gc_handle = rcl_guard_condition_get_rmw_handle(&guard_condition);
   });
   EXPECT_NE(nullptr, gc_handle);
+  // Test failed rcl_trigger_guard_condition
```

Comment:
This test is already getting pretty long, and `rcl_trigger_guard_condition` hasn't yet been tested in this function, what do you think about moving it to a separate test function?

Reply:
Done 639a17d